### ISSUE TITLE
Social (Publicize): Improve error message "Something went wrong. Please don't be mad"

### ISSUE THREAD

**Issue by devNigel** [Automattic/wp-calypso#58391]
### Quick summary

This error message generated in Publicize is not helpful and ironically it is these kinds of error messages that can make people mad.

The error was fixed by reconnecting Facebook. It would be more helpful to both user and Happiness Engineers if the error message suggested that in the first place.

Screenshot:

![image](https://user-images.githubusercontent.com/33957570/143010860-8b8deb0a-2803-442a-bb5f-ffa5ec6b56d6.png)

### Steps to reproduce

Not sure how to intentionally reproduce the error.

1. Go to Tools > Marketing > Connections > Facebook > Connect.
2. Go to Posts > Share post > and try to share the post on Facebook. If something goes wrong, you will see this error message.

### What you expected to happen

The error message should be helpful. If the details of the error cannot be provided, it should at least refer them to contact the support or suggest the next step for debugging.

### What actually happened

The error message does not help the user in fixing the issue.

### Context

Chat: 32872336-hc 

### Simple, Atomic or both?

Atomic

### Reproducibility

Intermittent

**Comment by Robertght** [comment]
@mmtr is this a change related to the recent Republicize changes?

**Comment by mmtr** [comment]
> @mmtr is this a change related to the recent Republicize changes?

I don't think so, I'm not aware of any change in how we notify this kind of errors. @pablinos or @retrofox can confirm

**Comment by Robertght** [comment]
Similar case: https://github.com/Automattic/wp-calypso/issues/28766

**Comment by benchilcote** [comment]
Reporting another case: #4622907-zen

User is attempting to republicize a post which is approximately 1 year old.  They have attempted to disconnect and reconnect to their Facebook account.  After doing so, they are still getting this error message:

"Something went wrong. Please don't be mad"

Also, I had the same feeling as @devNigel  - I think the error message could be improved.

**Comment by jartes** [comment]
I had another case on 33582346-hc

We tried to reconnect a couple of times all the connections (Twitter, Facebook, and Twitter) and didn't work. 

Finally, enabling all connections to all users made the trick (they were only enabled for that user).

**Comment by desnum** [comment]
Another case on 33675550-hc

**Comment by pablinos** [comment]
Looking at the code it seems this has been the error message for at least the last 5 years, so probably when the feature was first built. A version of it was introduced in #14550 for example.

It could be improved. I'm not sure if we have better information at this point, so suggesting that they reconnect with Facebook, or contact support, as @devNigel suggests, might be appropriate.

**Comment by cuemarie** [comment]
Ran into this vague error message here as well: 5351965-zd-woothemes

Tried lots of troublehshooting steps to no avail; thanks to @jartes comment, I've asked the user to try enabling this setting as well. They are the user that set up the FB connection, but the post has a different author.

Not sure if this error message only happens in this kind of situation, but clearer error messages could have definitely helped if so!

**Comment by wiesenhauss** [comment]
Another case in 39895755-hc
Enabling connections to all users worked.

**Comment by formosattic** [comment]
+1 6567895-zd-woothemes, happens with Instagram when media shared is a PNG but Instagram only accepts JPG/JPEG ([see doc](https://jetpack.com/support/jetpack-social/sharing-to-instagram-with-jetpack-social/)).

**Comment by jamiepalatnik** [comment]
+1 to editing this error message.
8477873-zd-a8c, the user received the error because they hit their max publish limit, so "Please try again later." would be appropriate in this case, or "Contact support for more details."
